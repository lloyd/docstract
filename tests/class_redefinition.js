/** It's an error to declare the same class twice */

/** @class Foo */
/** @endclass */
/** @class Foo */
/** @endclass */
