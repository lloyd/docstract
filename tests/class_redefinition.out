{
  "args": [
    "class 'Foo' redefined at line 5", 
    []
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, data, lineStart, line)"
    ], 
    [
      "_analyzeBlock", 
      "parseData['blockHandler'].merge(doc, data, guessedName)"
    ], 
    [
      "merge", 
      "raise RuntimeError(\"class '%s' redefined\" % doc[\"name\"])"
    ]
  ]
}
